pkg/diagnose/agentaccountcheck/agentaccountcheck_windows.go

```diff
@@ -0,0 +1,129 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+//go:build windows
+
+package agentaccountcheck
+
+import (
+	"fmt"
+	"strings"
+
+	diagnose "github.com/DataDog/datadog-agent/comp/core/diagnose/def"
+	"github.com/DataDog/datadog-agent/pkg/util/winutil"
+)
+
+// diagnoseImpl provides the Windows-specific implementation for agent account check diagnosis
+func diagnoseImpl() []diagnose.Diagnosis {
+	var diagnoses []diagnose.Diagnosis
+
+	// Check agent account groups
+	actualGroups, hasDesiredGroups, err := winutil.DoesAgentUserHaveDesiredGroups()
+	diagnoses = append(diagnoses, createGroupsDiagnosis(actualGroups, hasDesiredGroups, err))
+
+	// Check agent account rights
+	actualRights, hasDesiredRights, err := winutil.DoesAgentUserHaveDesiredRights()
+	diagnoses = append(diagnoses, createRightsDiagnosis(actualRights, hasDesiredRights, err))
+
+	return diagnoses
+}
+
+// createGroupsDiagnosis creates a diagnosis for agent account group membership
+func createGroupsDiagnosis(actualGroups []string, hasDesired bool, err error) diagnose.Diagnosis {
+	name := "Agent Account Group Membership"
+	category := "agent-account-check"
+	requiredGroups := []string{"Event Log Readers", "Performance Log Users", "Performance Monitor Users"}
+
+	if err != nil {
+		if isAccessDenied(err) {
+			return diagnose.Diagnosis{
+				Status:      diagnose.DiagnosisWarning,
+				Name:        name,
+				Diagnosis:   fmt.Sprintf("Cannot verify agent account group membership due to insufficient privileges.\n  Expected: %v\n  Detected: unable to check due to access denied", requiredGroups),
+				Category:    category,
+				RawError:    err.Error(),
+				Remediation: "Run as Administrator to check agent account group membership, or ensure the agent service is running with appropriate privileges.",
+			}
+		}
+		return diagnose.Diagnosis{
+			Status:      diagnose.DiagnosisUnexpectedError,
+			Name:        name,
+			Diagnosis:   fmt.Sprintf("Failed to check agent account group membership.\n  Expected: %v\n  Detected: error occurred", requiredGroups),
+			Category:    category,
+			RawError:    err.Error(),
+			Remediation: "Check if the agent is running with appropriate permissions.",
+		}
+	}
+
+	if hasDesired {
+		return diagnose.Diagnosis{
+			Status:    diagnose.DiagnosisSuccess,
+			Name:      name,
+			Diagnosis: "Agent account has all required group memberships",
+			Category:  category,
+		}
+	}
+
+	// Agent account doesn't have all required groups
+	return diagnose.Diagnosis{
+		Status:      diagnose.DiagnosisFail,
+		Name:        name,
+		Diagnosis:   fmt.Sprintf("Agent account is missing required group memberships.\n  Expected: %v\n  Detected: %v", requiredGroups, actualGroups),
```

Comment:
I do not think we need to print actualGroup, we can simple say which required group membership is missing